typed-store/src/traits.rs

```diff
@@ -54,4 +54,18 @@ where
 
     /// Returns a vector of values corresponding to the keys provided.
     fn multi_get(&self, keys: &[K]) -> Result<Vec<Option<V>>, Self::Error>;
+
+    /// Inserts key-value pairs atomically.
+    fn multi_insert<J, U>(
+        &self,
+        key_val_pairs: impl IntoIterator<Item = (J, U)>,
+    ) -> Result<(), Self::Error>
+    where
+        J: Borrow<K>,
+        U: Borrow<V>;
+
+    /// Removes keys atomically.
+    fn multi_remove<J>(&self, keys: impl IntoIterator<Item = J>) -> Result<(), Self::Error>
+    where
+        J: Borrow<K>;
 }
```

Comment:
So, we both agree that the `rocksDB` way of doing multi-inserts is the most efficient.

But there are many DBs that do not have atomic multi-inserts (e.g. ledgerDB ), so it might b unfair to request one such implementation in the trait. Could we:
- remove the comment on atomicity, or even better mention that atomicity isn't guaranteed, but some implementers may provide it?
- provide a [default implementation](https://doc.rust-lang.org/book/ch10-02-traits.html#default-implementations) for the multi-insert here, that would serve as a fallback for those DBs without the functionality?

Reply:
Right. Will remove the atomicity claim.